src/reconciliation/query.ts

```diff
@@ -0,0 +1,58 @@
+import {Catalog, IRI} from '@netwerk-digitaal-erfgoed/network-of-terms-catalog';
+import {QueryTermsService, Terms} from '../services/query';
+import {QueryMode} from '../search/query-mode';
+
+/**
+ * Fan out reconciliation query batch to terminology source queries.
+ *
+ * This first approach sends the source queries serially, in order to limit load on the sources.
+ * We may want to make this smarter by batching queries and acting on Timeout responses.
+ */
+export async function reconciliationQuery(
+  distributionIri: IRI,
+  query: ReconciliationQueryBatch,
+  catalog: Catalog,
+  queryTermsService: QueryTermsService
+): Promise<ReconciliationResultBatch> {
+  const distribution =
+    catalog.getDatasetByDistributionIri(distributionIri)!.distributions[0];
+
+  return await Object.entries(query).reduce(
+    async (
+      resultsPromise: Promise<ReconciliationResultBatch>,
+      [queryId, {query: queryString}]
+    ) => {
+      const results = await resultsPromise;
+      const termsResult = await queryTermsService.search(
+        queryString,
+        QueryMode.OPTIMIZED,
+        distribution,
+        10000
+      );
+      const terms = termsResult instanceof Terms ? termsResult.terms : [];
+      results[queryId] = {
+        result: terms.map(term => ({
+          id: term.id.value.toString(),
+          name: term.prefLabels.map(label => label.value).join(' • '), // Join similarly to network-of-terms-demo.
```

Comment:
Does this make sense? How do we want prefLabels to be shown in the reconciliation service?

Reply:
Good one. Our data model allows for more than one prefLabel (which could be handy if we want to start supporting multiple languages). But I'm not sure if there is a source that actually has 2+ prefLabels for a given term (in the same language). Anyway, does it make sense to return all the prefLabels - if any - in the `name`? Yes, I'd say so: if a source has multiple prefLabels, there's probably a reason for this. The `•` as a separator is interesting. I would prefer something more standardized (CSV?) so that the value of `name` could be parsed by a client, if desired. Perhaps `; `? But we'll run into problems if a prefLabel contains this separator :)